src/IotHub/IotHub/help/Get-AzIotHubMessageEnrichment.md

```diff
@@ -0,0 +1,170 @@
+﻿---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.IotHub.dll-Help.xml
+Module Name: Az.IotHub
+online version: https://docs.microsoft.com/en-us/powershell/module/az.iothub/get-aziothubmessageenrichment
+schema: 2.0.0
+---
+
+# Get-AzIotHubMessageEnrichment
+
+## SYNOPSIS
+Lists all message enrichments or a particular message enrichment for your IoT Hub.
+
+## SYNTAX
+
+### ResourceSet (Default)
+```
+Get-AzIotHubMessageEnrichment [-ResourceGroupName] <String> [-Name] <String> [-Key <String>]
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+### InputObjectSet
+```
+Get-AzIotHubMessageEnrichment [-InputObject] <PSIotHub> [-Key <String>]
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+### ResourceIdSet
+```
+Get-AzIotHubMessageEnrichment [-ResourceId] <String> [-Key <String>] [-DefaultProfile <IAzureContextContainer>]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
+For a detailed explanation of message enrichments in Azure IoT Hub, see https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-message-enrichments-overview
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\>  Get-AzIotHubMessageEnrichment -ResourceGroupName "myresourcegroup" -Name "myiothub"
+
+Key  Value   Endpoint(s)
+---  -----   -----------
+key1 value1  {endpoint1, endpoint2}
+key2 value2  {endpoint3, endpoint4}
+```
+
+List all message enrichments in MyIotHub
+
+### Example 2
+```powershell
+PS C:\>  Get-AzIotHubMessageEnrichment -ResourceGroupName "myresourcegroup" -Name "myiothub" -Key "newKey"
+
+Key         : key1
```

Comment:
This should either show the table format, or you can pipe the cmdlet above into Format-List in the example code, as in:

``` powershell
Get-AzIotHubMessageEnrichment -ResourceGroupName "myresourcegroup" -Name "myiothub" -Key "newKey" | Format-List

Reply:
The output is in table-format if there is more than one record otherwise it will show as list-format.

Reply:
Why?  That is not the recommended  pattern